Documentation for S3 sync rules (#1936)
elastic/connectors: docs/sync-rules/s3.md

### Setting up the Amazon S3 connector

See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.

### Example advanced sync rules

#### Indexing files and folders recursively

```json
[
  {
    "bucket": "bucket1",

>     "prefix": "folder1/folder2", # this will fetch files starting with folder2 and files in folder2

Reviewer: As an example, if you look at our elastic docs, you can see that the key names are defined above the sample code, with detailed explanations on what they do. The sample code is then comment-free.
https://www.elastic.co/guide/en/elasticsearch/reference/8.8/indices-create-index.html

### Setting up the Amazon S3 connector

See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.

### Overview

> The advanced sync rules consist of sets of rules, each having one or more parameters:
> 
> 1. bucket: Name of Amazon S3 bucket for which the rule would execute.
> 2. extension(optional): List of file extensions to be added for the sync. Defaults to empty list (i.e.sync all file extensions).
> 3. prefix(optional): Prefix is a string of characters at the beginning of the object key name, this will index files and folder data that matches with prefix value. Defaults to ""(i.e. sync all bucket objects).

Reviewer 2: ```suggestion
Advanced Sync Rules help manage data in Amazon S3 buckets. They take the following parameters:

1. `bucket`: This is the Amazon S3 bucket the rule applies to.
2. `extension` (optional): Lists which file types to sync. Defaults to syncing all types.
3. `prefix` (optional): String of prefix characters. The connector will index files and folder data that matches the string. Defaults to `""` (i.e. sync all bucket objects).
```

> ### Example advanced sync rules

Reviewer 2: ```suggestion
### Advanced sync rules examples
```

> ### Indexing files and folders recursively by specifying prefix

Reviewer 2: ```suggestion
### Indexing files and folders recursively by prefix
```

> #### Case1: Fetch files/folder starting with docs in folder1. Use '/' to fetch from subdirectory

Reviewer 2: ```suggestion
#### Example 
Fetch files/folders in `folder1/docs`.
```

> #### Case1: Fetch files/folder starting with docs. Use '/' to fetch from subdirectory

Reviewer 3: ```suggestion
#### Case1: Fetch files/folder starting with docs in folder1. Use '/' to fetch from subdirectory
```

```json
[
  {
    "bucket": "bucket1",
    "prefix": "folder1/docs"
  }

]

```

> #### Case2: Fetch files/folder starting with folder1

Reviewer 2: ```suggestion
#### Example

Fetch files/folder starting with `folder1`.
```

```json
[
  {
    "bucket": "bucket2",
    "prefix": "folder1"
  }
]

```

### Indexing files and folders by specifying extensions

> #### Case1: Fetch all objects which starts with `abc` and then filter using extension

Reviewer 2: ```suggestion
#### Example

Fetch all objects which start with `abc` and then filter using file extensions.
```
